src/lightning/fabric/utilities/distributed.py

```diff
@@ -23,9 +28,66 @@ class group:  # type: ignore
         WORLD = None
 
 
+if TYPE_CHECKING:
+    from lightning.fabric.strategies import Strategy
+
+
 log = logging.getLogger(__name__)
 
 
+@lru_cache(maxsize=128)
+def is_shared_filesystem(strategy: "Strategy", path: Optional[_PATH] = None, timeout: int = 3) -> bool:
+    """Checks whether the filesystem under the given path is shared across all processes.
+
+    Args:
+        strategy: The strategy being used, either from Fabric (``fabric.strategy``) or from Trainer
```

Comment:
Are these Fabric attributes?

Reply:
Both Trainer and Fabric have the `.strategy` attribute yes.